src/main/java/at/hannibal2/skyhanni/features/misc/FrozenTreasureTracker.kt

```diff
@@ -25,13 +25,13 @@ class FrozenTreasureTracker {
     private var display = emptyList<List<Any>>()
     private var estimatedIce = 0L
     private var lastEstimatedIce = 0L
-    private val icePerMin = mutableListOf<Long>()
+    private val icePerSecond = mutableListOf<Long>()
     private var icePerHour = 0
     private var stoppedChecks = 0
     private var compactPattern = "COMPACT! You found an Enchanted Ice!".toPattern()
 
     init {
-        fixedRateTimer(name = "skyhanni-dungeon-milestone-display", period = 1000) {
+        fixedRateTimer(name = "skyhanni-frozen-treasure-tracker", period = 1000) {
```

Comment:
A fixed rate timer can cause a ConcurrentModificationException crash if you swap worlds right as the timer ticks.

Reply:
My original idea was to use the LorenzTickEvent.repeatSecond(1) but this means that if the game lags and isn't at 20tps then the pelts/hr plus other parts of the tracker stop working. Right now it just has the try, catch but if you know a better way to do this lmk

Reply:
The better way would be to have exclusivity on the data. Only one thread can modify or access that objects content. You could achieve that by using `icePerSecond = mutableListOf()` inside of the onworldchange method